doc/tree.md

```diff
@@ -540,27 +554,36 @@ $tree = $repo->buildTree($query->getArrayResult(), $options);
 ```php
 <?php
 $controller = $this;
-        $tree = $root->childrenHierarchy(null,false,array('decorate' => true,
-            'rootOpen' => function($tree) {
-                if(count($tree) && ($tree[0]['lvl'] == 0)){
-                        return '<div class="catalog-list">';
-                }
-            },
-            'rootClose' => function($child) {
-                if(count($child) && ($child[0]['lvl'] == 0)){
-                                return '</div>';
-                }
-             },
-            'childOpen' => '',
-            'childClose' => '',
-            'nodeDecorator' => function($node) use (&$controller) {
-                if($node['lvl'] == 1) {
-                    return '<h1>'.$node['title'].'</h1>';
-                }elseif($node["isVisibleOnHome"]) {
-                    return '<a href="'.$controller->generateUrl("wareTree",array("id"=>$node['id'])).'">'.$node['title'].'</a>&nbsp;';
-                }
-            }
-        ));
+$tree = $root->childrenHierarchy(null, false, [
+    'decorate' => true,
+    'rootOpen' => static function (array $tree): ?string {
+        if ([] !== $tree && 0 == $tree[0]['lvl']) {
+            return '<div class="catalog-list">';
+        }
+
+        return null;
+    },
+    'rootClose' => static function (array $child): ?string {
+        if ([] !== $child && 0 == $child[0]['lvl']) {
+            return '</div>';
+        }
+
+        return null;
+     },
+    'childOpen' => '',
+    'childClose' => '',
+    'nodeDecorator' => static function(array $node) use (&$controller): ?string {
```

Comment:
```suggestion
    'nodeDecorator' => static function (array $node) use (&$controller): ?string {
```

Reply:
what about using [PHP-cs-fixer](https://cs.symfony.com/) ?

Reply:
> what about using [PHP-cs-fixer](https://cs.symfony.com/) ?

We can't apply PHP-CS-Fixer directly in this snippet as this file is using the Markdown format because it's part of the documentation.